domains/certificates/Query.API/API/KeyIssuer/KeyHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using API.KeyIssuer.Repositories;
+using Marten.Exceptions;
+using NSec.Cryptography;
+
+namespace API.KeyIssuer;
+
+internal class KeyHandler : IKeyIssuer
+{
+    private readonly SignatureAlgorithm algorithm;
+    private readonly IKeyIssuingRepository repository;
+
+    public KeyHandler(IKeyIssuingRepository repository)
+    {
+        algorithm = SignatureAlgorithm.Ed25519;
+        this.repository = repository;
+    }
+
+    public async Task<string> Create(string meteringPointOwner, CancellationToken cancellationToken)
```

Comment:
It seems to me that this method has several responsibilities.
I would suggest having a method to create the key, save it in the documentstore and one to fetch the publickey

Reply:
I think it is a clear violation of the SOLID principle and I believe we should strive after using that principle.
This will result in some small methods, but it will result in clean code that is easy to maintain

Reply:
I think it is kinda "hidden" that the Create method for a keyIssuer is returning the public key